src/shared/install.c

```diff
@@ -1337,7 +1337,7 @@ static int unit_file_load_or_readlink(
         if (r < 0)
                 return r;
 
-        if (path_equal(target, "/dev/null"))
+        if (null_or_empty_path(target) > 0)
```

Comment:
This doesn't seem to be correct either, because it doesn't take `root_dir` into account, so it'll resolve symlinks against the host.